test/replication-luatest/gh_7318_box_wait_limbo_acked_test.lua

```diff
@@ -0,0 +1,433 @@
+    end)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 1)
+    cg.master:exec(function()
+        box.cfg{
+            election_mode='off',
+            replication_synchro_quorum=1,
+        }
+    end)
+    local ff = require('fiber').create(function()
+        cg.replica:exec(function(f)
+            box.error.injection.set('ERRINJ_WAL_DELAY', false)
+            require('fiber').find(f):join()
+        end, {f})
+    end)
+    ff:set_joinable(true)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 0)
+    cg.replica_proxy:resume()
+    local ok, err = ff:join()
+    t.assert_equals(err, nil)
+    t.assert(ok)
+    server_becomes_the_leader_again(cg.master)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 0)
+end
+
+g.test_quorum_less_replication_synchro_quorum = function(cg)
+    local f = cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        local f = require('fiber').create(function() box.ctl.promote() end)
+        f:set_joinable(true)
+        return f:id()
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+    cg.master:exec(function()
+        require('fiber').create(function() box.space.test:insert{1} end)
+    end)
+    cg.replica:exec(function()
+        box.cfg{replication_synchro_quorum=1}
+    end)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 1)
+    local csw = get_fiber_csw(cg.replica, f)
+    cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+    end)
+    server_wait_fiber_csw_geq_than(cg.replica, f, csw + 1)
+    cg.replica:exec(function()
+        box.cfg{replication_synchro_quorum=2}
+    end)
+    csw = get_fiber_csw(cg.replica, f)
+    cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+    end)
+    server_wait_fiber_csw_geq_than(cg.replica, f, csw + 1)
+    cg.replica:exec(function(f)
+        box.cfg{replication_synchro_quorum=1}
+        require('fiber').find(f):join()
+    end, {f})
+    server_becomes_the_leader_again(cg.master)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 0)
+end
+
+g.test_new_synchronous_transactions_appeared_while_wait_quorum = function(cg)
+    cg.replica_proxy:pause()
+    cg.master:exec(function()
+        box.cfg{replication_synchro_quorum=3}
+    end)
+    local f = cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        local f = require('fiber').create(function() box.ctl.promote() end)
+        f:set_joinable(true)
+        return f:id()
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+    cg.master:exec(function()
+        require('fiber').create(function() box.space.test:insert{1} end)
+    end)
+    server_wait_synchro_queue_len_is_equal(cg.master, 1)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 1)
+    cg.master_proxy:pause()
```

Comment:
This test really passes almost every time.

My idea was that `proxy:pause()` doesn't work right away. So I did this:
```diff
@@ -404,6 +406,11 @@ g.test_new_synchronous_transactions_appeared_while_wait_quorum = function(cg)
     server_wait_synchro_queue_len_is_equal(cg.master, 1)
     server_wait_synchro_queue_len_is_equal(cg.replica, 1)
     cg.master_proxy:pause()
+    t.helpers.retrying({timeout=5}, function() cg.replica:exec(function()
+        local status = box.info.replication[1].upstream.status
+        require'log'.info(require'json'.encode(box.info.replication))
+        t.assert(status ~= 'follow')
+    end) end)
     cg.master:exec(function()
         require('fiber').create(function() box.space.test:insert{2} end)
     end)
```

Now the test started failing when master waits for synchro queue len to become 2. Master's logs contain:
```
2023-10-11 12:47:07.503 [39726] main/111/applier/unix/:/tmp/t/001_replication-luatest/               replica_proxy.sock I> can't read row
  61 2023-10-11 12:47:07.503 [39726] main/111/applier/unix/:/tmp/t/001_replication-luatest/               replica_proxy.sock coio.c:296 E> TimedOut: timed out
  62 2023-10-11 12:47:07.503 [39726] main/111/applier/unix/:/tmp/t/001_replication-luatest/               replica_proxy.sock I> will retry every 0.10 second
  63 2023-10-11 12:47:07.503 [39726] main/111/applier/unix/:/tmp/t/001_replication-luatest/               replica_proxy.sock raft.c:1221 W> RAFT: resigning from leader state
  64 2023-10-11 12:47:07.564 [39726] relay/unix/:(socket)/101/main coio.c:296 E> TimedOut: timed out
  65 2023-10-11 12:47:07.564 [39726] relay/unix/:(socket)/101/main I> exiting the relay loop
  66 2023-10-11 12:47:07.596 [39726] main/124/lua box.cc:355 E> ER_READONLY: Can't modify data on a       read-only instance - state is election follower with term 2
  67 2023-10-11 12:47:07.596 [39726] main/124/lua box.cc:355 E> ER_READONLY: Can't modify data on a       read-only instance - state is election follower with term 2
```

It seems master steps down due to fencing. For some reason setting `election_fencing_mode = 'off'` on master didn't help, but I didn't dig into it much.